### DISCUSSION TITLE
[WebSearch][GooglePSE] Bad request due invalid query-param

### DISCUSSION THREAD

**Discussion by mluksenberg** [open-webui/open-webui#7256] [Issues]
# Bug Report
When I try to use GooglePSE as web engine for searching, it is returning bad request due the param &num=30 is not allowed by Google. The fix is not add that query-param for that specific engine
Here is an example how it is building the URL: https://www.googleapis.com/customsearch/v1?cx={iD}&q=Test&key={MYQUEY}&num=30
And Here is the correct URL: https://www.googleapis.com/customsearch/v1?cx={iD}&q=Test&key={MYQUEY}

**Comment by tjbck** [comment]
https://developers.google.com/custom-search/v1/reference/rest/v1/cse/list#request
